### ISSUE TITLE
Nets pretrained with batch normalization

### ISSUE THREAD

**Issue by nightrome** [vlfeat/matconvnet#530]
I am experiencing terrible performance with the imagenet-matconvnet-vgg-verydeep-16 net. With imagenet-vgg-verydeep-16 it works fine.
It says in [1] that they are trained in MatConvNet (beta17) and with batch normalization. Why are the batch normalization layers not included? That seems to be the reason for my terrible performance. I think there should at least be a visible warning that these networks don't work as-is.

[1] http://www.vlfeat.org/matconvnet/pretrained/

**Comment by vedaldi** [comment]
Hi, could you clarify whether this was a problem in fine-tuning or the network itself was preferably badly?

These are deployed models, so BN is absorbed in convolutional layers for speed. However, it is true that for fine-tuning the BN layers are best left alone. I'll add a warning to the documentation and possibly include also the non-deployed models.

**Comment by nightrome** [comment]
Thanks. Yes it was in finetuning.

**Comment by nightrome** [comment]
Would it be possible to provide the non-deployed model with batch normalization? I think estimating the mean and variation parameters of each layer manually is pretty tricky.

**Comment by nightrome** [comment]
Okay, found a possible workaround here..
https://github.com/vlfeat/matconvnet/issues/533
